submission-receipt/main.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+  Copyright (C) 2022,  icgc-argo
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  Authors:
+    Edmund Su
+"""
+
+import argparse
+import requests
+import json
+import csv
+import sys
+import json
+
+def main():
+    """
+    Python implementation of tool: sanity-check
+    This is auto-generated Python code, please update as needed!
+    """
+
+    parser = argparse.ArgumentParser(description='Tool: sanity-check')
+    parser.add_argument('-s', '--study_id', dest='study_id',type=str,
+                        help='Submission SONG API interaction URL', required=True)
+    parser.add_argument('-a', '--analysis_id', dest='analysis_id',type=str,
+                        help='Submission SONG API interaction URL', required=True)
+    parser.add_argument('-u', '--submission_song_url', dest='submission_song_url',type=str,
+                        help='Submission SONG API interaction URL', required=True)
+    parser.add_argument('-f', '--files', dest='files',nargs='+',type=str,
+                        help='Submission SONG API interaction URL', required=True)
+    parser.add_argument('-o', '--output_file', dest='output_file',type=str,
+                        help='Submission SONG API interaction URL', required=True)
+    args = parser.parse_args()
+
+    header={"accept":"*/*"}
+    url=args.submission_song_url+"/studies/%s/analysis/%s" % (args.study_id,args.analysis_id)
+    try:
+        response=requests.get(url,headers=header)
+    except requests.exceptions.RequestException as e:  # This is the correct syntax
+        print("Unable to establish connection")
+        raise SystemExit(e)
+
+    if response.status_code==200:
+        with open(args.output_file, 'w', newline='') as csvfile:
+            metadata={
+                "analysisState":None,
+                'publishedAt':None,
+                'submitterSampleId':None,
+                'submitterSpecimenId':None,
+                'submitterDonorId':None,
+                'sampleId':None,
+                'specimenId':None,
+                "donorId":None,
+                "analysisId":None,
+                "studyId":None,
+                "objectId":None,
+                "fileName":None,
+                "fileMd5sum":None
+                }
```

Comment:
Do we consider the scenario when the submitted files are original CRAMs?

Reply:
When the users submitted `CRAM`, we can add additional fields `original_fileName`, `original_fileSize` and `original_fileMd5sum` since the user only have info for their submitted `CRAM` for cross check.